driver_go18_test.go

```diff
@@ -189,27 +160,40 @@ func testCancel(dbt *DBTest, ctx context.Context, cancel context.CancelFunc, que
 	}
 
 	errChan := make(chan error)
-	go checkCancellation(dbt.db, dbname, query, cancel, errChan, getIdTimeout, killTimeout, pollTimeout)
-
-	// This query will be canceled.
-	startTime := time.Now()
-	err = queryFunc()
-	if err != nil && err != context.Canceled {
-		errLog.Print(err)
-	}
+	go func() {
+		// This query will be canceled.
+		startTime := time.Now()		
+		err = queryFunc()
+		if err != nil && err != context.Canceled {
+			errLog.Print(err)
+		}
+		if err != context.Canceled && ctx.Err() != context.Canceled {
+			errChan <- fmt.Errorf("expected context.Canceled, got %v", err)
+		}
+		if d := time.Since(startTime); d > 500*time.Millisecond {
```

Comment:
I am not sure, that need to check this.
your timeouts for get query and waiting while kill query, should cover all cases.

Reply:
removed